File: backend/models/Word.js
const mongoose = require('mongoose');

const wordSchema = new mongoose.Schema({
  userId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true
  },
  word: {
    type: String,
    required: true,
    trim: true
  },
  // 拼音 (for Chinese words)
  pinyin: {
    type: String
  },
  // 英文翻译
  translation: {
    type: String
  },
  definition: {
    type: String
  },
  examples: [{
    type: String
  }],
  // 难度级别
  difficulty: {
    type: String,
    enum: ['beginner', 'intermediate', 'advanced'],
    default: 'intermediate'
  },
  status: {
    type: String,
    enum: ['unknown', 'known'],
    default: 'unknown'
  },
  sourceImage: {
    type: String // URL to the image in S3
  },
  addedAt: {
    type: Date,
    default: Date.now
  },
  learnedAt: {
    type: Date
  },
  reviewCount: {
    type: Number,
    default: 0
  },
  lastReviewedAt: {
    type: Date
  },
  nextReviewAt: {
    type: Date
  }
}, { timestamps: true });

// Compound index for userId and word
wordSchema.index({ userId: 1, word: 1 }, { unique: true });

module.exports = mongoose.model('Word', wordSchema);

File: backend_new/core/models/word.py
"""Word model - same schema as Node/Mongoose backend."""
from django.db import models
from django_mongodb_backend.fields import EmbeddedModelArrayField, ObjectIdField
from django_mongodb_backend.models import EmbeddedModel


class CompoundEmbed(EmbeddedModel):
    word = models.CharField(max_length=100)
    pinyin = models.CharField(max_length=200, blank=True)
    meaning = models.CharField(max_length=300, blank=True)


class ExampleEmbed(EmbeddedModel):
    chinese = models.TextField()
    pinyin = models.CharField(max_length=500, blank=True)
    english = models.CharField(max_length=500, blank=True)


class Word(models.Model):
    userId = ObjectIdField()
    word = models.CharField(max_length=100)
    pinyin = models.CharField(max_length=200, blank=True)
    translation = models.CharField(max_length=500, blank=True)
    definition = models.CharField(max_length=500, blank=True)
    compounds = EmbeddedModelArrayField(CompoundEmbed, default=list, blank=True)
    examples = EmbeddedModelArrayField(ExampleEmbed, default=list, blank=True)
    difficulty = models.CharField(
        max_length=20,
        choices=[("beginner", "beginner"), ("intermediate", "intermediate"), ("advanced", "advanced")],
        default="intermediate",
    )
    status = models.CharField(
        max_length=20,
        choices=[("unknown", "unknown"), ("known", "known")],
        default="unknown",
    )
    sourceImage = models.CharField(max_length=500, blank=True)
    addedAt = models.DateTimeField(auto_now_add=True, null=True, blank=True)
    learnedAt = models.DateTimeField(null=True, blank=True)
    reviewCount = models.IntegerField(default=0)
    lastReviewedAt = models.DateTimeField(null=True, blank=True)
    nextReviewAt = models.DateTimeField(null=True, blank=True)

    class Meta:
        db_table = "words"
        managed = False

    def __str__(self):
        return self.word
